lib/services.js

```diff
@@ -410,57 +410,35 @@ const services = {
             if (mpuBucket.getMdBucketModelVersion() < 2) {
                 splitter = constants.oldSplitter;
             }
-            const splitterLen = splitter.length;
-            const searchArgs = {
-                prefix: `overview${splitter}${objectKey}${splitter}${uploadId}`,
-                marker: undefined,
-                delimiter: undefined,
-                maxKeys: 1,
-            };
-            metadata.listObject(mpuBucketName, searchArgs,
-                log, (err, response) => {
+            const mpuOverviewKey =
+                `overview${splitter}${objectKey}${splitter}${uploadId}`;
+
+            metadata.getObjectMD(mpuBucket.getName(), mpuOverviewKey,
+                {}, log, (err, storedMetadata) => {
                     if (err) {
+                        if (err.NoSuchKey) {
+                            return cb(errors.NoSuchUpload);
+                        }
                         log.error('error from metadata', { error: err });
                         return cb(err);
                     }
-                    if (response.Contents.length !== 1) {
-                        return cb(errors.NoSuchUpload);
-                    }
-
-                    // Add other mpu info to the mpuOverview Array
-                    const storedValue = response.Contents[0].value;
-                    const eventualStorageBucket = storedValue
-                        .eventualStorageBucket;
-                    const initiatorID = storedValue.Initiator.ID;
-                    const initiatorDisplayName = storedValue
-                        .Initiator.DisplayName;
-                    const ownerID = storedValue.Owner.ID;
-                    const ownerDisplayName = storedValue.Owner.DisplayName;
-                    const storageClass = storedValue.StorageClass;
-                    const initiated = storedValue.Inititated;
 
-                    // Take the saved overview and convert it into an array
-                    // so can pull info from name
-                    const key = response.Contents[0].key;
-                    // key form:
-                    // - overview
-                    // - {splitter}
-                    // - {objectKey}
-                    // - {splitter}
-                    // - {uploadId}
-                    const index = key.indexOf(splitter);
-                    const index2 = key.lastIndexOf(splitter);
+                    const initiatorID = storedMetadata.initiator.ID;
+                    const ownerID = storedMetadata['owner-id'];
                     const mpuOverview = [
-                        key.substring(0, index),
-                        key.substring(index + splitterLen, index2),
-                        key.substring(index2 + splitterLen),
-                        eventualStorageBucket,
+                        // 'overview' left as placeholder in array for
+                        // backwards compatibility
```

Comment:
If we can change this into an object from an array without too much trouble (too many code changes in other api's), we should do so.